pkg/templates/pdbmaxunavailable/template.go

```diff
@@ -29,24 +33,40 @@ func init() {
 		ParseAndValidateParams: params.ParseAndValidate,
 		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
 			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
-
 				// Get the PDB provided
 				pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
 				if !ok {
 					return nil
 				}
 
-				// If MaxUnavailable is set to 0 (both int and string)
-				// return as no other checks will uniquely apply
-				if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntVal == 0 {
-					return []diagnostic.Diagnostic{{
-						Message: maxUnavailableZeroMsg,
-					},
+				if pdb.Spec.MaxUnavailable == nil {
+					if pdb.Spec.MaxUnavailable.IntVal == 0 {
+						return []diagnostic.Diagnostic{{
+							Message: "MaxUnavailable is undefined",
```

Comment:
I think this is "undesired" diagnostic. I mean we would like to report when `MaxUnavailable` is 0, otherwise do nothing.

Reply:
still have a bit of mixed feeling on this one but a "missing" maxUnavailable is valid. skipping validation on nil. please take a look

Reply:
yes it's valid. Thanks. Looks good.